### PR TITLE
Utilize vector operations over scalar if possible

### LINKED ISSUE #4085: Prefer vector operations over unary operations for various GLSL/SPIRV/HLSL functions
Prefer vector operations over unary operations for `roundEven` and `isfinite`.

### PR DESCRIPTION
**Pull request by jkwak-work**
Closes #4085

### CODE REVIEW

File: source/slang/hlsl.meta.slang [hunk 1]
```diff
@@ -4888,7 +4888,17 @@ uint16_t asuint16(float16_t value)
 [__readNone]
 [require(cuda_glsl_hlsl_spirv, shader5_sm_5_0)]
 vector<uint16_t,N> asuint16<let N : int>(vector<float16_t,N> value)
-{ VECTOR_MAP_UNARY(uint16_t, N, asuint16, value); }
+{
+    __target_switch
+    {
+    case hlsl: __intrinsic_asm "asuint16";
+    case spirv: return spirv_asm {
+        OpBitcast $$vector<uint16_t,N> result $value
```
Comment (csyonghe) [inline source/slang/hlsl.meta.slang:4896]:
```
result:$$vector<uint16_t,N> = OpBitcast %value
```

Comment (jkwak-work) [reply source/slang/hlsl.meta.slang:4896]: I thought they are the same thing.
Is it more preferable style to use `result:$$XX = `?

**Comment by jkwak-work** [comment]
It looks like the change in `f16tof32()` is causing a test failure.
```
// half-conversion.slang
//TEST:CROSS_COMPILE:-target spirv-assembly -entry main -stage fragment

cbuffer C
{
        uint4 u;
}

float4 main() : SV_Target
{
        return f16tof32(u);
}
```
I think we are deprecating "CROSS_COMPILE" mode.
And this test may not be valid anymore.
If it is still valid, I am gonna have to revert the changes in `f16tof32()`.

**Comment by jkwak-work** [comment]
There are more tests failing.
I should have tested before creating this PR.
Sorry about that.

I am gonna turn this into "draft" until it is ready for a review.

**Comment by csyonghe** [comment]
We should convert all failing `CROSS_COMPILE` into `SIMPLE` test with filecheck rules.

If we keep doing this, then eventually all cross_compile tests will be gone.

File: source/slang/hlsl.meta.slang (hunk 1 above)
Comment (csyonghe) [reply source/slang/hlsl.meta.slang:4896]: Yes, this is more preferred.

**Review by ArielG-NV — COMMENTED** [review]
Looks good to me

**Review by csyonghe — APPROVED** [review]

**Pull request merged**